### ISSUE TITLE
Security issues, Remote Command Execution Vulnerability

### ISSUE THREAD

**Issue by WangYihang** [Codiad/Codiad#1017]
hey , I found a vulnerability which effect all versions yesterday , attackers are able to execute any system command on the target server which has codiad running , more detail , email please , wangyihanger@gmail.com

**Comment by cheiff** [comment]
Replied to you privately. Thank you.

**Comment by WangYihang** [comment]
:100: OK , I just have received Mr.Javi's email , I will tell him the detail of this vulnerability :D

**Comment by Magic-Zero** [comment]
hey, maybe i found an error in the file.if you repair it, attackers are allowed to upload arbitrary file and execute any code by this file.if you need provide more details, email please
magiczero@protonmail.com

**Comment by WangYihang** [comment]
:+1:  Cool , good good!

**Comment by cheiff** [comment]
Hi @WangYihang ,

Demo is fixed using this new PR. Please confirm that it is solved.
Thanks

@Magic-Zero 
Please, create a new issue for this. It is unrelated with the OP.

Thank you, best

**Comment by WangYihang** [comment]
Hey, I am testing this vulnerability again, but I found that the code on demo.codiad.com seems to be the different of the code on github. I failed on the demo.codiad.com but successed  exploit on the code of github

**Comment by cheiff** [comment]
Yes, that is because the PR has not yet been accepted. @daeks has to look at it.

**Comment by WangYihang** [comment]
Oh~ I am sorry , I forgot to inform you whether this vulnerbility is solved
In my opinion , I cannot exploit anymore , I think you can merge the patch
to the master branch now.

2017-08-27 16:36 GMT+08:00 Javi <notifications@github.com>:

> Yes, that is because the PR has not yet been accepted. @daeks has to look
> at it.
>

**Comment by daeks** [comment]
It is already accepted but tbh Codiad is currently not really maintained anymore. I havent seen or heard anything from Kent (@Fluidbyte) since a very long time and personally I am not using Codiad anymore. 
From time to time, I am testing some PRs with my legacy installation and merging them, but that is all..,

**Comment by cheiff** [comment]
We keep using it and I would be glad to be able to continue maintaining it.

Would that be possible?

**Comment by daeks** [comment]
As it still belongs to Kent, it is his decision, but anyhow it lives from its contributions. Everyone is able to provide changes by pull requests and I would really appreciate if someone volunteers to test and review them (github has this functionallity). I can still merge it afterwards as it is not so important anymore who finally presses the button to merge.

**Comment by cheiff** [comment]
OK. I'll try to do my best to keep this alive. 
Many thanks.

**Comment by aggsol** [comment]
So, is the vulnerability now fixed? I am about to install it soon...

**Comment by marclebrun** [comment]
It's very sad that Codiad is not maintainted anymore...
Do you know of any other online editor as great as this one ?
Thanks :-)

**Comment by 4oo4** [comment]
Wow, can't believe I missed this. Definitely need to find something else to switch to.
